structuralcodes/sections/_generic.py

```diff
@@ -901,22 +1038,96 @@ def _compute_ultimate_strain_profiles(self, theta: float = 0):
+        type5: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type6: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
     ) -> s_res.NMMInteractionDomain:
-        """Calculates the NMM interaction domain."""
+        """Calculates the NMM interaction domain.
+
+        Args:
+        num_theta (int) Number of discretization of angle of NA
+            (Optional, Default = 32)
+        n1: (int) number of strain profiles in field 1 (Optional, default = 1)
+        n2: (int) number of strain profiles in field 2 (Optional, default = 2)
+        n3: (int) number of strain profiles in field 3 (Optional, default = 15)
+        n4: (int) number of strain profiles in field 4 (Optional, default = 10)
+        n5: (int) number of strain profiles in field 5 (Optional, default = 3)
+        n6: (int) number of strain profiles in field 6 (Optional, default = 4)
+        n: (int) total number of strain profiles (Optional, default = None)
+            If specified n and n1..n6 the total number of n may be different.
+        type1: (literal) type of spacing for field 1
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type2: (literal) type of spacing for field 2
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type3: (literal) type of spacing for field 3
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'geometric')
+        type4: (literal) type of spacing for field 4
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type5: (literal) type of spacing for field 5
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type6: (literal) type of spacing for field 6
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+
+        Return:
+        (NMMInteractionDomain)
+        """
         res = s_res.NMMInteractionDomain()
         res.num_theta = num_theta
-        # TODO: if we want to resample and structure better data we can
-        # use n_axial as the number of discretizations in axial direction
-        res.num_axial = num_axial
+
+        # Process n
+        if n is not None:
+            n1_attempt = int(n / 35 * 1)
+            n2_attempt = int(n / 35 * 2)
+            n3_attempt = int(n / 35 * 15)
+            n4_attempt = int(n / 35 * 10)
+            n5_attempt = int(n / 35 * 3)
+            n6_attempt = int(n / 35 * 4)
+            n1 = max(n1_attempt, n1)
+            n2 = max(n2_attempt, n2)
+            n3 = max(n3_attempt, n3)
+            n4 = max(n4_attempt, n4)
+            n5 = max(n5_attempt, n5)
+            n6 = max(n6_attempt, n6)
```

Comment:
I don't immediately see what is going on here. Please consider adding a sentence in the comment on line 1095 to explain.

Reply:
I have factored out these lines into a function since it was used twice. I documented it properly hopefully.